File: internal/content/testdata/bench/csharp/nc-median-two-sorted.cs
public class Solution {
    public long find_median_sorted_arrays(long[] nums1, long[] nums2) {
        if (nums1.Length > nums2.Length) {
            var tmp = nums1; nums1 = nums2; nums2 = tmp;
        }
        int m = nums1.Length, n = nums2.Length;
        int half = (m + n) / 2;
        int lo = 0, hi = m;
        while (true) {
            int i = (lo + hi) / 2;
            int j = half - i;
            long left1  = i > 0 ? nums1[i - 1] : long.MinValue;
            long right1 = i < m ? nums1[i]     : long.MaxValue;
            long left2  = j > 0 ? nums2[j - 1] : long.MinValue;
            long right2 = j < n ? nums2[j]     : long.MaxValue;
            if (left1 <= right2 && left2 <= right1) {
                if ((m + n) % 2 == 1)
                    return Math.Min(right1, right2);
                return (long)((Math.Max(left1, left2) + Math.Min(right1, right2)) / 2.0);
            } else if (left1 > right2) {
                hi = i - 1;
            } else {
                lo = i + 1;
            }
        }
    }
}

File: internal/content/testdata/bench/javascript/nc-median-two-sorted.js
function find_median_sorted_arrays(nums1, nums2) {
    if (nums1.length > nums2.length) {
        [nums1, nums2] = [nums2, nums1];
    }
    const m = nums1.length, n = nums2.length;
    const half = Math.floor((m + n) / 2);
    let lo = 0, hi = m;
    while (true) {
        const i = Math.floor((lo + hi) / 2);
        const j = half - i;
        const left1  = i > 0 ? nums1[i - 1] : -Infinity;
        const right1 = i < m ? nums1[i]     :  Infinity;
        const left2  = j > 0 ? nums2[j - 1] : -Infinity;
        const right2 = j < n ? nums2[j]     :  Infinity;
        if (left1 <= right2 && left2 <= right1) {
            if ((m + n) % 2 === 1) {
                return Math.min(right1, right2);
            }
            return Math.trunc((Math.max(left1, left2) + Math.min(right1, right2)) / 2);
        } else if (left1 > right2) {
            hi = i - 1;
        } else {
            lo = i + 1;
        }
    }
}
